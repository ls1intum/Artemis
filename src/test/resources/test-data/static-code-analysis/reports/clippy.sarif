{
  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0.json",
  "runs": [
    {
      "results": [
        {
          "level": "error",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 34,
                  "byteOffset": 327,
                  "endColumn": 6,
                  "endLine": 23,
                  "startColumn": 5,
                  "startLine": 21
                }
              }
            }
          ],
          "message": {
            "text": "usage of an `unsafe` block"
          },
          "relatedLocations": [],
          "ruleId": "unsafe_code",
          "ruleIndex": 0
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 8,
                  "byteOffset": 557,
                  "endColumn": 21,
                  "endLine": 42,
                  "startColumn": 13,
                  "startLine": 42
                }
              }
            }
          ],
          "message": {
            "text": "almost complete ascii range"
          },
          "relatedLocations": [
            {
              "message": {
                "text": "use an inclusive range \"..=\""
              },
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 2,
                  "byteOffset": 560,
                  "endColumn": 18,
                  "endLine": 42,
                  "startColumn": 16,
                  "startLine": 42
                }
              }
            }
          ],
          "ruleId": "clippy::almost_complete_range",
          "ruleIndex": 1
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 14,
                  "byteOffset": 402,
                  "endColumn": 19,
                  "endLine": 28,
                  "startColumn": 5,
                  "startLine": 28
                }
              }
            }
          ],
          "message": {
            "text": "unreachable statement"
          },
          "relatedLocations": [],
          "ruleId": "unreachable_code",
          "ruleIndex": 2
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 6,
                  "byteOffset": 235,
                  "endColumn": 10,
                  "endLine": 15,
                  "startColumn": 4,
                  "startLine": 15
                }
              }
            }
          ],
          "message": {
            "text": "function `unused` is never used"
          },
          "relatedLocations": [],
          "ruleId": "dead_code",
          "ruleIndex": 3
        },
        {
          "level": "error",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 5,
                  "byteOffset": 517,
                  "endColumn": 18,
                  "endLine": 38,
                  "startColumn": 13,
                  "startLine": 38
                }
              }
            }
          ],
          "message": {
            "text": "this operation will always return zero. This is likely not the intended outcome"
          },
          "relatedLocations": [],
          "ruleId": "clippy::erasing_op",
          "ruleIndex": 4
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 8,
                  "byteOffset": 594,
                  "endColumn": 13,
                  "endLine": 46,
                  "startColumn": 5,
                  "startLine": 46
                }
              }
            }
          ],
          "message": {
            "text": "unneeded `return` statement"
          },
          "relatedLocations": [
            {
              "message": {
                "text": "remove `return` \"1\""
              },
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 8,
                  "byteOffset": 594,
                  "endColumn": 13,
                  "endLine": 46,
                  "startColumn": 5,
                  "startLine": 46
                }
              }
            },
            {
              "message": {
                "text": "remove `return` \"1\" \"\""
              },
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 1,
                  "byteOffset": 602,
                  "endColumn": 14,
                  "endLine": 46,
                  "startColumn": 13,
                  "startLine": 46
                }
              }
            }
          ],
          "ruleId": "clippy::needless_return",
          "ruleIndex": 5
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 85,
                  "byteOffset": 607,
                  "endColumn": 86,
                  "endLine": 49,
                  "startColumn": 1,
                  "startLine": 49
                }
              }
            }
          ],
          "message": {
            "text": "this function has too many arguments (8/7)"
          },
          "relatedLocations": [],
          "ruleId": "clippy::too_many_arguments",
          "ruleIndex": 6
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 43,
                  "byteOffset": 714,
                  "endColumn": 48,
                  "endLine": 53,
                  "startColumn": 5,
                  "startLine": 53
                }
              }
            }
          ],
          "message": {
            "text": "large array defined as const"
          },
          "relatedLocations": [
            {
              "message": {
                "text": "make this a static item \"static\""
              },
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 5,
                  "byteOffset": 714,
                  "endColumn": 10,
                  "endLine": 53,
                  "startColumn": 5,
                  "startLine": 53
                }
              }
            }
          ],
          "ruleId": "clippy::large_const_arrays",
          "ruleIndex": 7
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 32,
                  "byteOffset": 804,
                  "endColumn": 8,
                  "endLine": 59,
                  "startColumn": 5,
                  "startLine": 58
                }
              }
            }
          ],
          "message": {
            "text": "needless use of `for_each`"
          },
          "relatedLocations": [
            {
              "message": {
                "text": "try \"for _x in v.iter() {\n    }\""
              },
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 32,
                  "byteOffset": 804,
                  "endColumn": 8,
                  "endLine": 59,
                  "startColumn": 5,
                  "startLine": 58
                }
              }
            }
          ],
          "ruleId": "clippy::needless_for_each",
          "ruleIndex": 8
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 7,
                  "byteOffset": 863,
                  "endColumn": 12,
                  "endLine": 63,
                  "startColumn": 5,
                  "startLine": 63
                }
              }
            }
          ],
          "message": {
            "text": "`todo` should not be present in production code"
          },
          "relatedLocations": [],
          "ruleId": "clippy::todo",
          "ruleIndex": 9
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 16,
                  "byteOffset": 251,
                  "endColumn": 20,
                  "endLine": 17,
                  "startColumn": 4,
                  "startLine": 17
                }
              }
            }
          ],
          "message": {
            "text": "function `nonStandardStyle` should have a snake case name"
          },
          "relatedLocations": [
            {
              "message": {
                "text": "convert the identifier to snake case \"non_standard_style\""
              },
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 16,
                  "byteOffset": 251,
                  "endColumn": 20,
                  "endLine": 17,
                  "startColumn": 4,
                  "startLine": 17
                }
              }
            }
          ],
          "ruleId": "non_snake_case",
          "ruleIndex": 10
        },
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 8,
                  "byteOffset": 458,
                  "endColumn": 13,
                  "endLine": 33,
                  "startColumn": 5,
                  "startLine": 33
                }
              }
            }
          ],
          "message": {
            "text": "calls to `std::mem::drop` with a reference instead of an owned value does nothing"
          },
          "relatedLocations": [
            {
              "message": {
                "text": "use `let _ = ...` to ignore the expression or result \"let _ = \""
              },
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 5,
                  "byteOffset": 458,
                  "endColumn": 10,
                  "endLine": 33,
                  "startColumn": 5,
                  "startLine": 33
                }
              }
            },
            {
              "message": {
                "text": "use `let _ = ...` to ignore the expression or result \"let _ = \" \"\""
              },
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "src/main.rs"
                },
                "region": {
                  "byteLength": 1,
                  "byteOffset": 465,
                  "endColumn": 13,
                  "endLine": 33,
                  "startColumn": 12,
                  "startLine": 33
                }
              }
            }
          ],
          "ruleId": "dropping_references",
          "ruleIndex": 11
        }
      ],
      "tool": {
        "driver": {
          "informationUri": "https://rust-lang.github.io/rust-clippy/",
          "name": "clippy",
          "rules": [
            {
              "fullDescription": {
                "text": "requested on the command line with `-F unsafe-code`\n"
              },
              "id": "unsafe_code"
            },
            {
              "fullDescription": {
                "text": "for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_complete_range\n`#[warn(clippy::almost_complete_range)]` on by default\n"
              },
              "helpUri": "https://rust-lang.github.io/rust-clippy/master/index.html#almost_complete_range",
              "id": "clippy::almost_complete_range"
            },
            {
              "fullDescription": {
                "text": "`#[warn(unreachable_code)]` on by default\n"
              },
              "id": "unreachable_code"
            },
            {
              "fullDescription": {
                "text": "`#[warn(dead_code)]` on by default\n"
              },
              "id": "dead_code"
            },
            {
              "fullDescription": {
                "text": "for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#erasing_op\n`#[deny(clippy::erasing_op)]` on by default\n"
              },
              "helpUri": "https://rust-lang.github.io/rust-clippy/master/index.html#erasing_op",
              "id": "clippy::erasing_op"
            },
            {
              "fullDescription": {
                "text": "for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return\n`#[warn(clippy::needless_return)]` on by default\n"
              },
              "helpUri": "https://rust-lang.github.io/rust-clippy/master/index.html#needless_return",
              "id": "clippy::needless_return"
            },
            {
              "fullDescription": {
                "text": "for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments\n`#[warn(clippy::too_many_arguments)]` on by default\n"
              },
              "helpUri": "https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments",
              "id": "clippy::too_many_arguments"
            },
            {
              "fullDescription": {
                "text": "for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#large_const_arrays\n`#[warn(clippy::large_const_arrays)]` on by default\n"
              },
              "helpUri": "https://rust-lang.github.io/rust-clippy/master/index.html#large_const_arrays",
              "id": "clippy::large_const_arrays"
            },
            {
              "fullDescription": {
                "text": "for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_for_each\n`-W clippy::needless-for-each` implied by `-W clippy::pedantic`\nto override `-W clippy::pedantic` add `#[allow(clippy::needless_for_each)]`\n"
              },
              "helpUri": "https://rust-lang.github.io/rust-clippy/master/index.html#needless_for_each",
              "id": "clippy::needless_for_each"
            },
            {
              "fullDescription": {
                "text": "for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#todo\nrequested on the command line with `-W clippy::todo`\n"
              },
              "helpUri": "https://rust-lang.github.io/rust-clippy/master/index.html#todo",
              "id": "clippy::todo"
            },
            {
              "fullDescription": {
                "text": "`#[warn(non_snake_case)]` on by default\n"
              },
              "id": "non_snake_case"
            },
            {
              "fullDescription": {
                "text": "`#[warn(dropping_references)]` on by default\n"
              },
              "id": "dropping_references"
            }
          ]
        }
      }
    }
  ],
  "version": "2.1.0"
}