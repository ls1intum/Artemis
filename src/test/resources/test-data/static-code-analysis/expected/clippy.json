{
  "tool": "CLIPPY",
  "issues": [
    {
      "filePath": "src/main.rs",
      "startLine": 21,
      "endLine": 23,
      "startColumn": 5,
      "endColumn": 6,
      "rule": "unsafe_code",
      "category": "unsafe_code",
      "message": "usage of an `unsafe` block",
      "priority": "error"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 42,
      "endLine": 42,
      "startColumn": 13,
      "endColumn": 21,
      "rule": "clippy::almost_complete_range",
      "category": "suspicious",
      "message": "almost complete ascii range",
      "priority": "warning"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 28,
      "endLine": 28,
      "startColumn": 5,
      "endColumn": 19,
      "rule": "unreachable_code",
      "category": "unused",
      "message": "unreachable statement",
      "priority": "warning"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 15,
      "endLine": 15,
      "startColumn": 4,
      "endColumn": 10,
      "rule": "dead_code",
      "category": "unused",
      "message": "function `unused` is never used",
      "priority": "warning"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 38,
      "endLine": 38,
      "startColumn": 13,
      "endColumn": 18,
      "rule": "clippy::erasing_op",
      "category": "correctness",
      "message": "this operation will always return zero. This is likely not the intended outcome",
      "priority": "error"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 46,
      "endLine": 46,
      "startColumn": 5,
      "endColumn": 13,
      "rule": "clippy::needless_return",
      "category": "style",
      "message": "unneeded `return` statement",
      "priority": "warning"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 49,
      "endLine": 49,
      "startColumn": 1,
      "endColumn": 86,
      "rule": "clippy::too_many_arguments",
      "category": "complexity",
      "message": "this function has too many arguments (8/7)",
      "priority": "warning"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 53,
      "endLine": 53,
      "startColumn": 5,
      "endColumn": 48,
      "rule": "clippy::large_const_arrays",
      "category": "perf",
      "message": "large array defined as const",
      "priority": "warning"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 58,
      "endLine": 59,
      "startColumn": 5,
      "endColumn": 8,
      "rule": "clippy::needless_for_each",
      "category": "pedantic",
      "message": "needless use of `for_each`",
      "priority": "warning"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 63,
      "endLine": 63,
      "startColumn": 5,
      "endColumn": 12,
      "rule": "clippy::todo",
      "category": "restriction",
      "message": "`todo` should not be present in production code",
      "priority": "warning"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 17,
      "endLine": 17,
      "startColumn": 4,
      "endColumn": 20,
      "rule": "non_snake_case",
      "category": "nonstandard_style",
      "message": "function `nonStandardStyle` should have a snake case name",
      "priority": "warning"
    },
    {
      "filePath": "src/main.rs",
      "startLine": 33,
      "endLine": 33,
      "startColumn": 5,
      "endColumn": 13,
      "rule": "dropping_references",
      "category": "rust_other",
      "message": "calls to `std::mem::drop` with a reference instead of an owned value does nothing",
      "priority": "warning"
    }
  ]
}
